lib/std/jsbigints.nim

```diff
@@ -0,0 +1,159 @@
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asUintN
+  runnableExamples:
+    doAssert toUint(32, newJsBigInt"2147483647") == newJsBigInt"2147483647"
+
+func toInt*(a: JsBigInt): int {.importjs: "Number(#)".} =
+  runnableExamples:
+    doAssert toInt(newJsBigInt"2147483647") == 2147483647
+
+func `+`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".} =
+  runnableExamples:
+    doAssert (newJsBigInt"9" + newJsBigInt"1") == newJsBigInt"10"
+
+func `-`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".} =
+  runnableExamples:
+    doAssert (newJsBigInt"9" - newJsBigInt"1") == newJsBigInt"8"
+
+func `*`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".} =
+  runnableExamples:
+    doAssert (newJsBigInt"42" * newJsBigInt"9") == newJsBigInt"378"
+
+func `div`*(x, y: JsBigInt): JsBigInt {.importjs: "(# / #)".} =
+  ## Same as `div` but for `JsBigInt`(uses JavaScript `BigInt() / BigInt()`).
+  runnableExamples:
+    doAssert newJsBigInt"13" div newJsBigInt"3" == newJsBigInt"4"
+    doAssert newJsBigInt"-13" div newJsBigInt"3" == newJsBigInt"-4"
+    doAssert newJsBigInt"13" div newJsBigInt"-3" == newJsBigInt"-4"
+    doAssert newJsBigInt"-13" div newJsBigInt"-3" == newJsBigInt"4"
+
+func `mod`*(x, y: JsBigInt): JsBigInt {.importjs: "(# % #)".} =
+  ## Same as `mod` but for `JsBigInt` (uses JavaScript `BigInt() % BigInt()`).
+  runnableExamples:
+    doAssert newJsBigInt"13" mod newJsBigInt"3" == newJsBigInt"1"
+    doAssert newJsBigInt"-13" mod newJsBigInt"3" == newJsBigInt"-1"
+    doAssert newJsBigInt"13" mod newJsBigInt"-3" == newJsBigInt"1"
+    doAssert newJsBigInt"-13" mod newJsBigInt"-3" == newJsBigInt"-1"
+
+func `<`*(x, y: JsBigInt): bool {.importjs: "(# $1 #)".} =
+  runnableExamples:
+    doAssert newJsBigInt"2" < newJsBigInt"9"
+
+func `<=`*(x, y: JsBigInt): bool {.importjs: "(# $1 #)".} =
+  runnableExamples:
+    doAssert newJsBigInt"1" <= newJsBigInt"5"
+
+func `==`*(x, y: JsBigInt): bool {.importjs: "(# === #)".} =
+  runnableExamples:
+    doAssert newJsBigInt"42" == newJsBigInt"42"
+
+func `**`*(x, y: JsBigInt): JsBigInt {.importjs: "((#) $1 #)".} =
+  runnableExamples:
+    doAssert (newJsBigInt"9" ** newJsBigInt"5") == newJsBigInt"59049"
+
+func `xor`*(x, y: JsBigInt): JsBigInt {.importjs: "(# ^ #)".} =
+  runnableExamples:
+    doAssert (newJsBigInt"555" xor newJsBigInt"2") == newJsBigInt"553"
+
+func `shl`*(a, b: JsBigInt): JsBigInt {.importjs: "(# << #)".} =
+  runnableExamples:
+    doAssert (newJsBigInt"999" shl newJsBigInt"2") == newJsBigInt"3996"
+
+func `shr`*(a, b: JsBigInt): JsBigInt {.importjs: "(# >> #)".} =
+  runnableExamples:
+    doAssert (newJsBigInt"999" shr newJsBigInt"2") == newJsBigInt"249"
+
+func `-`*(a: JsBigInt): JsBigInt {.importjs: "($1#)".} =
+  runnableExamples:
+    doAssert -(newJsBigInt"10101010101") == newJsBigInt"-10101010101"
+
+func inc*(a: var JsBigInt; b = 1) {.importjs: "([#][0][0] += BigInt(#))".} =
```

Comment:
the previous (IIRC) way was better, inc should have a separate overload with just 1 arg so it can use a more efficient implementation in js (incrementing a bigint vs adding 2 bigints)

`func inc*(a: var JsBigInt) {.importjs: "([#][0][0]++".} = ...`